Dockerfile

```diff
@@ -1,38 +1,86 @@
 ARG VERSION=unspecified
+ARG PY_VERSION=3.10.0
 
-FROM python:3.10.0-alpine
+FROM python:${PY_VERSION}-slim-bullseye
 
 ARG VERSION
 
 # For a list of pre-defined annotation keys and value types see:
 # https://github.com/opencontainers/image-spec/blob/master/annotations.md
 # Note: Additional labels are added by the build workflow.
-LABEL org.opencontainers.image.authors="mark.feldhousen@cisa.dhs.gov"
+LABEL org.opencontainers.image.authors="jeremy.frasier@cisa.dhs.gov"
 LABEL org.opencontainers.image.vendor="Cybersecurity and Infrastructure Security Agency"
 
-ARG CISA_UID=421
+# Manually set up the virtual environment
+ENV PY_VENV=/.venv
+RUN python -m venv ${PY_VENV}
+ENV PATH="${PY_VENV}/bin:$PATH"
+RUN python -m pip install --upgrade pip
+
+###
+# Setup the unprivileged user and its home directory
+###
+ARG CISA_GID=421
+ARG CISA_UID=${CISA_GID}
+ENV CISA_USER="cisa"
+ENV CISA_GROUP=${CISA_USER}
 ENV CISA_HOME="/home/cisa"
-ENV ECHO_MESSAGE="Hello World from Dockerfile"
 
-RUN addgroup --system --gid ${CISA_UID} cisa \
-  && adduser --system --uid ${CISA_UID} --ingroup cisa cisa
+###
+# Create unprivileged user
+###
+RUN groupadd --system --gid ${CISA_GID} ${CISA_GROUP} && \
+    useradd --system --uid ${CISA_UID} --gid ${CISA_GROUP} --comment "${CISA_USER} user" ${CISA_USER}
+
+###
+# Install everything we need
+###
 
-RUN apk --update --no-cache add \
-ca-certificates \
-openssl \
-py-pip
+###
+# Dependencies
+#
+# We need redis-tools so we can use redis-cli to communicate with
+# redis.  wget is used inside of gather-domains.sh.
+#
+# Install dependencies are only needed for software installation and
```

Comment:
I have a current project (not ⚔️ 🐉) where I did something very similar.  I think it would be a useful example: 

https://github.com/felddy/weewx-docker/blob/develop/Dockerfile

Reply:
I had the same thought when creating the `Dockerfile`, but at the time I was trying to get something going that I could test.  I shall make this change.

Reply:
@cisagov/team-ois - please have a look at commit cf86400 and let me know what you think.

Reply:
Wunderbar!